Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -1684,90 +1840,557 @@ RefPtr<StyleValue> Parser::parse_css_value(ParsingContext const& context, Proper
+    // '+'?† <ndashdigit-ident>
+    // In all of these cases, the + is optional, and has no effect.
+    // So, we just skip the +, and carry on.
+    if (!is_delim(first_value, "+"sv)) {
         values.reconsume_current_input_token();
+        // We do *not* skip whitespace here.
     }
 
-    current_value = values.next_token();
+    auto& first_after_plus = values.next_token();
+    // '+'?† n
+    // '+'?† n <signed-integer>
+    // '+'?† n ['+' | '-'] <signless-integer>
+    if (is_n(first_after_plus)) {
+        a = 1;
+        values.skip_whitespace();
+        auto& second_value = values.next_token();
+        if (second_value.is(Token::Type::EndOfFile)) {
+            // '+'?† n
+            return make_return_value();
+        } else if (is_signed_integer(second_value)) {
+            // '+'?† n <signed-integer>
+            b = second_value.token().to_integer();
+            return make_return_value();
+        }
 
-    if (is_n(current_value)) {
         values.skip_whitespace();
+        auto& third_value = values.next_token();
+        if ((is_delim(second_value, "+"sv) || is_delim(second_value, "-"sv)) && is_signless_integer(third_value)) {
+            // '+'?† n ['+' | '-'] <signless-integer>
+            b = third_value.token().to_integer() * (is_delim(second_value, "+"sv) ? 1 : -1);
+            return make_return_value();
+        }
 
-        auto next_value = values.peek_token();
-        if (is_delim(next_value, "+") || is_delim(next_value, "-")) {
-            const auto sign = is_delim(next_value, "+") ? 1 : -1;
-            values.next_token();
+        return syntax_error();
+    }
 
-            values.skip_whitespace();
+    // '+'?† n- <signless-integer>
+    if (is_ndash(first_after_plus)) {
+        values.skip_whitespace();
+        auto& second_value = values.next_token();
+        if (is_signless_integer(second_value)) {
+            a = 1;
+            b = -second_value.token().to_integer();
+            return make_return_value();
+        }
 
-            // "An+B" pattern
-            auto number = values.next_token();
-            if (!number.is(Token::Type::Number))
-                return {};
+        return syntax_error();
+    }
 
-            pattern.step_size = step_size_or_offset;
-            pattern.offset = sign * number.token().integer();
-        } else {
-            // "An" pattern
-            pattern.step_size = step_size_or_offset;
+    // '+'?† <ndashdigit-ident>
+    if (is_ndashdigit_ident(first_after_plus)) {
+        a = 1;
+        auto maybe_b = first_after_plus.token().ident().substring_view(1).to_int();
+        if (maybe_b.has_value()) {
+            b = maybe_b.value();
+            return make_return_value();
         }
+
+        return syntax_error();
+    }
+
+    return syntax_error();
+}
+
+OwnPtr<CalculatedStyleValue::CalcSum> Parser::parse_calc_expression(ParsingContext const& context, Vector<StyleComponentValueRule> const& values)
+{
+    dbgln("Parsing calc()");
```

Comment:
`dbglnif` as all the other `dbgln`s? Or warp in`#ifdef` as in `parse_a_n_plus_b_pattern()`

Reply:
Please prefer `if constexpr (THING_DEBUG)` over the preprocessor way, that makes having code rot a lot harder :)

Reply:
Oops, that's leftover too. I got overexcited when I thought I was done. :sweat_smile:

There are a couple of cases in `parse_a_n_plus_b_pattern()` that were wrapped in an `#if`, so thanks for the tip @alimpfard !